sessiontxn/isolation/repeatable_read.go

```diff
@@ -179,24 +182,58 @@ func (p *PessimisticRRTxnContextProvider) AdviseOptimizeWithPlan(val interface{}
 		plan = execute.Plan
 	}
 
-	mayOptimizeForPointGet := false
-	if v, ok := plan.(*plannercore.PhysicalLock); ok {
-		if _, ok := v.Children()[0].(*plannercore.PointGetPlan); ok {
-			mayOptimizeForPointGet = true
+	p.optimizeForNotFetchingLatestTS = optimizeForNotFetchingLatestTS(plan, false)
+
+	return nil
+}
+
+// optimizeForNotFetchingLatestTS searches for optimization condition recursively
+// Note: For point get and batch point get (name it plan), if one of the ancestor node is update/delete/physicalLock,
+// we should check whether the plan.Lock is true or false. See comments in needNotToBeOptimized.
+// flag = true means one of the ancestor node is update/delete/physicalLock.
+func optimizeForNotFetchingLatestTS(plan plannercore.Plan, flag bool) bool {
```

Comment:
flag is meaningless, maybe `inWriteStmt` is better

Reply:
I have modified this method.